### ISSUE TITLE
🐛 lint/suspicious/noRedeclare breaks *.d.ts files

### ISSUE THREAD

**Issue by yckbilly1929** [biomejs/biome#137]
### Environment information

```block
CLI:
  Version:                      1.0.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.3.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.6.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0

Discovering running Biome servers...

Incompatible Biome Server: ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

i Rage discovered this running server using an incompatible version of Biome.

Server:
  Version:                      <=10.0.0
```

### What happened?

Not sure it is related to Rome or Biome. Error exists on a commonly pre-generated react types file.

![screencapture 2023-09-04 135257](https://github.com/biomejs/biome/assets/12884758/ee46c123-6fd2-4bb3-bf2d-e779abbbb380)

### Expected result

It should not throw an error. Or is there any configuration i missed?

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
